spring-cloud-data-module-deployers/spring-cloud-data-module-deployer-cloudfoundry/src/main/java/org/springframework/cloud/data/module/deployer/cloudfoundry/CloudFoundryModuleDeployer.java

```diff
@@ -16,32 +16,73 @@
 
 package org.springframework.cloud.data.module.deployer.cloudfoundry;
 
+import java.util.HashMap;
 import java.util.Map;
 
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.cloud.data.core.ModuleDeploymentId;
 import org.springframework.cloud.data.core.ModuleDeploymentRequest;
 import org.springframework.cloud.data.module.ModuleStatus;
 import org.springframework.cloud.data.module.deployer.ModuleDeployer;
+import org.springframework.stereotype.Component;
 
-public class CloudFoundryModuleDeployer implements ModuleDeployer {
+/**
+ * A {@link ModuleDeployer} which deploys modules as applications running in a space in CloudFoundry.
+ */
+@Component
+public final class CloudFoundryModuleDeployer implements ModuleDeployer {
+
+	private final CloudFoundryModuleDeploymentIdConverter cloudFoundryModuleDeploymentIdConverter;
+
+	private final CloudFoundryResourceClient resource;
+
+	@Autowired
+	public CloudFoundryModuleDeployer(CloudFoundryModuleDeploymentIdConverter cloudFoundryModuleDeploymentIdConverter,
+			CloudFoundryResourceClient resource) {
+		this.cloudFoundryModuleDeploymentIdConverter = cloudFoundryModuleDeploymentIdConverter;
+		this.resource = resource;
+	}
 
+	// TODO: implement deploy
 	@Override
 	public ModuleDeploymentId deploy(ModuleDeploymentRequest request) {
-		throw new UnsupportedOperationException();
+		throw new UnsupportedOperationException("to be implemented");
+		// return null;
 	}
 
 	@Override
-	public void undeploy(ModuleDeploymentId id) {
-		throw new UnsupportedOperationException();
+	public Map<ModuleDeploymentId, ModuleStatus> status() {
+		GetApplicationsStatusResponse response = this.resource.getApplicationsStatus(
+				new GetApplicationsStatusRequest());
```

Comment:
It returns all the application statuses for the *space*. There might be a way of paring it down if I can filter on a name pattern rather than just a name. But the api documentation is so bad I don't know how.